File: Components/BlogML/Xml/BlogMLPost.vb
Imports System.Xml.Serialization

Namespace BlogML.Xml
 <Serializable>
 Public NotInheritable Class BlogMLPost
  Inherits BlogMLNode

  <XmlAttribute("post-url")>
  Public Property PostUrl As String

  <XmlAttribute("hasexcerpt")>
  Public Property HasExcerpt As Boolean = False

  <XmlAttribute("type")>
  Public Property PostType As New BlogPostTypes

  <XmlAttribute("views")>
  Public Property Views As UInt32 = 0

  <XmlAttribute("image", [Namespace]:="http://dnn-connect.org/blog/")>
  Public Property Image As String

  <XmlAttribute("allow-comments", [Namespace]:="http://dnn-connect.org/blog/")>
  Public Property AllowComments As Boolean = True

  <XmlAttribute("display-copyright", [Namespace]:="http://dnn-connect.org/blog/")>
  Public Property DisplayCopyright As Boolean = False

  <XmlAttribute("copyright", [Namespace]:="http://dnn-connect.org/blog/")>
  Public Property Copyright As String

  <XmlAttribute("locale", [Namespace]:="http://dnn-connect.org/blog/")>
  Public Property Locale As String

  <XmlElement("post-name")>
  Public Property PostName As String

  <XmlElement("content")>
  Public Property Content As New BlogMLContent

  <XmlElement("excerpt")>
  Public Property Excerpt As New BlogMLContent

  <XmlArray("authors")>
  <XmlArrayItem("author", GetType(BlogMLAuthorReference))>
  Public ReadOnly Property Authors As New AuthorReferenceCollection

  <XmlArray("categories")>
  <XmlArrayItem("category", GetType(BlogMLCategoryReference))>
  Public ReadOnly Property Categories As New CategoryReferenceCollection

  <XmlArray("comments")>
  <XmlArrayItem("comment", GetType(BlogMLComment))>
  Public ReadOnly Property Comments As New CommentCollection

  <XmlArray("trackbacks")>
  <XmlArrayItem("trackback", GetType(BlogMLTrackback))>
  Public ReadOnly Property Trackbacks As New TrackbackCollection

  <XmlArray("attachments")>
  <XmlArrayItem("attachment", GetType(BlogMLAttachment))>
  Public ReadOnly Property Attachments As New AttachmentCollection

  <Serializable>
  Public NotInheritable Class AuthorReferenceCollection
   Inherits ArrayList
   Default Public Shadows ReadOnly Property Item(index As Integer) As BlogMLAuthorReference
    Get
     Return TryCast(MyBase.Item(index), BlogMLAuthorReference)
    End Get
   End Property

   Public Overloads Sub Add(value As BlogMLAuthorReference)
    MyBase.Add(value)
   End Sub

   Public Overloads Function Add(authorID As String) As BlogMLAuthorReference
    Dim item As New BlogMLAuthorReference
    item.Ref = authorID
    MyBase.Add(item)
    Return item
   End Function
  End Class

  <Serializable>
  Public NotInheritable Class CommentCollection
   Inherits ArrayList
   Default Public Shadows ReadOnly Property Item(index As Integer) As BlogMLComment
    Get
     Return TryCast(MyBase.Item(index), BlogMLComment)
    End Get
   End Property

   Public Overloads Sub Add(value As BlogMLComment)
    MyBase.Add(value)
   End Sub
  End Class

  <Serializable>
  Public NotInheritable Class TrackbackCollection
   Inherits ArrayList
   Default Public Shadows ReadOnly Property Item(index As Integer) As BlogMLTrackback
    Get
     Return TryCast(MyBase.Item(index), BlogMLTrackback)
    End Get
   End Property

   Public Overloads Sub Add(value As BlogMLTrackback)
    MyBase.Add(value)
   End Sub
  End Class

  <Serializable>
  Public NotInheritable Class CategoryReferenceCollection
   Inherits ArrayList
   Default Public Shadows ReadOnly Property Item(index As Integer) As BlogMLCategoryReference
    Get
     Return TryCast(MyBase.Item(index), BlogMLCategoryReference)
    End Get
   End Property

   Public Overloads Sub Add(value As BlogMLCategoryReference)
    MyBase.Add(value)
   End Sub

   Public Overloads Function Add(categoryID As String) As BlogMLCategoryReference
    Dim item As New BlogMLCategoryReference
    item.Ref = categoryID
    MyBase.Add(item)
    Return item
   End Function
  End Class

  <Serializable>
  Public NotInheritable Class AttachmentCollection
   Inherits ArrayList
   Default Public Shadows ReadOnly Property Item(index As Integer) As BlogMLAttachment
    Get
     Return TryCast(MyBase.Item(index), BlogMLAttachment)
    End Get
   End Property

   Public Overloads Sub Add(value As BlogMLAttachment)
    MyBase.Add(value)
   End Sub
  End Class
 End Class
End Namespace

File: Server/Core/BlogML/Xml/BlogMLPost.cs
using System;
using System.Collections;
using System.Xml.Serialization;

namespace DotNetNuke.Modules.Blog.Core.BlogML.Xml
{
  [Serializable]
  public sealed class BlogMLPost : BlogMLNode
  {

    [XmlAttribute("post-url")]
    public string PostUrl { get; set; }

    [XmlAttribute("hasexcerpt")]
    public bool HasExcerpt { get; set; } = false;

    [XmlAttribute("type")]
    public BlogPostTypes PostType { get; set; } = new BlogPostTypes();

    [XmlAttribute("views")]
    public uint Views { get; set; } = 0U;

    [XmlAttribute("image", Namespace = "http://dnn-connect.org/blog/")]
    public string Image { get; set; }

    [XmlAttribute("allow-comments", Namespace = "http://dnn-connect.org/blog/")]
    public bool AllowComments { get; set; } = true;

    [XmlAttribute("display-copyright", Namespace = "http://dnn-connect.org/blog/")]
    public bool DisplayCopyright { get; set; } = false;

    [XmlAttribute("copyright", Namespace = "http://dnn-connect.org/blog/")]
    public string Copyright { get; set; }

    [XmlAttribute("locale", Namespace = "http://dnn-connect.org/blog/")]
    public string Locale { get; set; }

    [XmlElement("post-name")]
    public string PostName { get; set; }

    [XmlElement("content")]
    public BlogMLContent Content { get; set; } = new BlogMLContent();

    [XmlElement("excerpt")]
    public BlogMLContent Excerpt { get; set; } = new BlogMLContent();

    [XmlArray("authors")]
    [XmlArrayItem("author", typeof(BlogMLAuthorReference))]
    public AuthorReferenceCollection Authors { get; set; } = new AuthorReferenceCollection();

    [XmlArray("categories")]
    [XmlArrayItem("category", typeof(BlogMLCategoryReference))]
    public CategoryReferenceCollection Categories { get; set; } = new CategoryReferenceCollection();

    [XmlArray("comments")]
    [XmlArrayItem("comment", typeof(BlogMLComment))]
    public CommentCollection Comments { get; set; } = new CommentCollection();

    [XmlArray("trackbacks")]
    [XmlArrayItem("trackback", typeof(BlogMLTrackback))]
    public TrackbackCollection Trackbacks { get; set; } = new TrackbackCollection();

    [XmlArray("attachments")]
    [XmlArrayItem("attachment", typeof(BlogMLAttachment))]
    public AttachmentCollection Attachments { get; set; } = new AttachmentCollection();

    [Serializable]
    public sealed class AuthorReferenceCollection : ArrayList
    {
      public new BlogMLAuthorReference this[int index]
      {
        get
        {
          return base[index] as BlogMLAuthorReference;
        }
      }

      public void Add(BlogMLAuthorReference value)
      {
        base.Add(value);
      }

      public BlogMLAuthorReference Add(string authorID)
      {
        var item = new BlogMLAuthorReference();
        item.Ref = authorID;
        base.Add(item);
        return item;
      }
    }

    [Serializable]
    public sealed class CommentCollection : ArrayList
    {
      public new BlogMLComment this[int index]
      {
        get
        {
          return base[index] as BlogMLComment;
        }
      }

      public void Add(BlogMLComment value)
      {
        base.Add(value);
      }
    }

    [Serializable]
    public sealed class TrackbackCollection : ArrayList
    {
      public new BlogMLTrackback this[int index]
      {
        get
        {
          return base[index] as BlogMLTrackback;
        }
      }

      public void Add(BlogMLTrackback value)
      {
        base.Add(value);
      }
    }

    [Serializable]
    public sealed class CategoryReferenceCollection : ArrayList
    {
      public new BlogMLCategoryReference this[int index]
      {
        get
        {
          return base[index] as BlogMLCategoryReference;
        }
      }

      public void Add(BlogMLCategoryReference value)
      {
        base.Add(value);
      }

      public BlogMLCategoryReference Add(string categoryID)
      {
        var item = new BlogMLCategoryReference();
        item.Ref = categoryID;
        base.Add(item);
        return item;
      }
    }

    [Serializable]
    public sealed class AttachmentCollection : ArrayList
    {
      public new BlogMLAttachment this[int index]
      {
        get
        {
          return base[index] as BlogMLAttachment;
        }
      }

      public void Add(BlogMLAttachment value)
      {
        base.Add(value);
      }
    }
  }
}